rpc/types/src/misc/tx_entry.rs

```diff
@@ -37,23 +37,118 @@ pub struct TxEntry {
-    tx_hash: String,
+/// [`EpeeObjectBuilder`] for [`TxEntry`].
+///
+/// Not for public usage.
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct __TxEntryEpeeBuilder {
+    pub as_hex: Option<String>,
+    pub as_json: Option<String>,
+    pub block_height: Option<u64>,
+    pub block_timestamp: Option<u64>,
+    pub confirmations: Option<u64>,
+    pub double_spend_seen: Option<bool>,
+    pub in_pool: Option<bool>,
+    pub output_indices: Option<Vec<u64>>,
+    pub prunable_as_hex: Option<String>,
+    pub prunable_hash: Option<String>,
+    pub pruned_as_hex: Option<String>,
+    pub received_timestamp: Option<u64>,
+    pub relayed: Option<bool>,
+    pub tx_hash: Option<String>,
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObjectBuilder<TxEntry> for __TxEntryEpeeBuilder {
+    fn add_field<B: Buf>(&mut self, name: &str, r: &mut B) -> error::Result<bool> {
+        macro_rules! read_epee_field {
+            ($($field:ident),*) => {
+                match name {
+                    $(
+                        stringify!($field) => { self.$field = Some(read_epee_value(r)?); },
+                    )*
+                    _ => return Ok(false),
+                }
+            };
+        }
+
+        read_epee_field! {
+            as_hex,
+            as_json,
+            block_height,
+            block_timestamp,
+            confirmations,
+            double_spend_seen,
+            in_pool,
+            output_indices,
+            prunable_as_hex,
+            prunable_hash,
+            pruned_as_hex,
+            received_timestamp,
+            relayed,
+            tx_hash
+        }
+
+        Ok(true)
+    }
+
+    fn finish(self) -> error::Result<TxEntry> {
+        const ELSE: error::Error = error::Error::Format("Required field was not found!");
+        Ok(TxEntry {
+            as_hex: self.as_hex.ok_or(ELSE)?,
+            as_json: self.as_json.ok_or(ELSE)?,
+            block_height: self.block_height.ok_or(ELSE)?,
+            block_timestamp: self.block_timestamp.ok_or(ELSE)?,
+            confirmations: self.confirmations.ok_or(ELSE)?,
+            double_spend_seen: self.double_spend_seen.ok_or(ELSE)?,
+            in_pool: self.in_pool.ok_or(ELSE)?,
+            output_indices: self.output_indices.ok_or(ELSE)?,
+            prunable_as_hex: self.prunable_as_hex.ok_or(ELSE)?,
+            prunable_hash: self.prunable_hash.ok_or(ELSE)?,
+            pruned_as_hex: self.pruned_as_hex.ok_or(ELSE)?,
+            received_timestamp: self.received_timestamp.ok_or(ELSE)?,
+            relayed: self.relayed.ok_or(ELSE)?,
+            tx_hash: self.tx_hash.ok_or(ELSE)?,
+        })
+    }
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObject for TxEntry {
+    type Builder = __TxEntryEpeeBuilder;
+
+    fn number_of_fields(&self) -> u64 {
+        // TODO: this is either 12 or 10 depending on `self.in_pool`.
+        14
+    }
```

Comment:
How should this work? We sometimes serialize 10 fields, sometimes 12.

Reply:
This can be dynamic, however it could be less than 10, epee wont serialize empty container.